doc/edi_in_depth.md

```diff
@@ -87,6 +87,7 @@ A full EDI schema `file_declaration` is as follows:
     "segment_delimiter": "<segment delimiter>",                     <== required
     "element_delimiter": "<element delimiter>",                     <== required
     "component_delimiter": "<component delimiter>",                 <== optional
+    "repetition_delimiter": "<repetition delimiter>",               <== optional
```

Comment:
[OUT OF SCOPE]

These 4 delimiters may be gleaned from the `ISA` header as the whole segment is fixed length and the separator characters are in deterministic positions:

```bash
ISA*00*          *00*          *ZZ*CMSFFM         *ZZ*987654321      *230614*1605*^*00501*000001003*1*T*:~
```

```bash
segment_delimiter      - index position 105
element_delimiter      - index position 4
component_delimiter.   - index position 104
repetition_delimiter   - index position 82
```

This is clearly beyond the scope of this change, and the proposed configuration option matches the current paradigm as expected.

Reply:
Thanks for the feedback. Yes we were aware of the fixed positions of these separators/delimiters in the header segment. However, our design goal / philosophy, as you might have figured out by now, was to make this library as flexible as possible and not bound to a given EDI standard. Different standards (X12/EDIFact/etc) have different headers, different delimiters and their positions. We felt the current approach allows flexibility.

Reply:
That's great to know. Thanks for sharing your insight.